cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/AbstractRuntimeTwillPreparer.java

```diff
@@ -773,7 +777,7 @@ private JvmOptions getJvmOptions() {
    * Returns the extra options for the container JVM.
    */
   private String addClassLoaderClassName(String extraOptions) {
-    if (classLoaderClassName == null) {
+    if (classLoaderClassName == null || TetheringProvisioner.TETHERING_NAME.equals(provisioner)) {
```

Comment:
why is this needed?

Reply:
Before this PR, TetheringRuntimeJobManager just copied over `app.program.jvm.opts` from cConf - https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java#L257

Now, we are setting `app.program.jvm.opts` in the TetheringLaunchMessage using the jvm opts added in the twill preparer.

Reply:
ok I see, it's used by DataprocJobMain and RemoteLauncher, which isn't used in the tethering flow. I don't think we actually need to skip this, but if we do, it should be done by skipping the call to `.setClassLoader()` on the TwillPreparer when there is a tethered run, instead of skipping adding it in this class. Otherwise we're relying on an implementation detail to avoid setting it rather than using the actual API.

Reply:
reverted this change.